File: js-pr/56/javascript/code-quality-readability-maintainability/data_processor_01.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function process(d) {
    if (d) {
        if (d.length > 0) {
            for (let i = 0; i < d.length; i++) {
                if (d[i]) {
                    if (typeof d[i] === 'string') {
                        if (d[i].length > 5) {
                            d[i] = d[i].substring(0, 5);
                        }
                    }
                }
            }
        }
    }
    return d;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/data_processor_01.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def process(d):
    if d:
        if len(d) > 0:
            for i in range(len(d)):
                if d[i]:
                    if isinstance(d[i], str):
                        if len(d[i]) > 5:
                            d[i] = d[i][:5]
    return d
# {/fact}